python/ray/data/_internal/plan.py

```diff
@@ -647,6 +647,17 @@ def execute(
                     reply.store_stats.restored_bytes_total
                 )
 
+            stats.dataset_bytes_spilled = 0
```

Comment:
let's avoid the use of the phrase "dataset" when possible (we are gradually moving from "Ray Dataset" to "Ray Data"), maybe byes_spilled_total or something similar?